ddl/table.go

```diff
@@ -53,29 +55,65 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 		}
 	}
 
-	ver, err = updateSchemaVersion(t, job)
-	if err != nil {
-		return ver, errors.Trace(err)
-	}
-
+	originalState := job.SchemaState
 	switch tbInfo.State {
 	case model.StateNone:
 		// none -> public
-		tbInfo.State = model.StatePublic
+		if withSelect {
+			tbInfo.State = model.StateWriteOnly
+		} else {
+			tbInfo.State = model.StatePublic
+		}
 		tbInfo.UpdateTS = t.StartTS
 		err = t.CreateTable(schemaID, tbInfo)
 		if err != nil {
+			job.State = model.JobStateCancelled
```

Comment:
Shouldn't set it to  model.JobStateCancelled. Because some error we can retry, like tikv busy.